### ISSUE TITLE
Imported code can mask symbols.

### ISSUE THREAD

**Issue by mbostock** [rollup/rollup#20]
Let’s say I have some code in foo.js:

```js
export var quince = 42;

export default function() {
  console.log("foo");
};
```

Meanwhile, over in bar.js:

```js
import foo from "./foo";

export default function() {
  foo();
  console.log("bar", quince);
};
```

(Note that bar.js is referring to `quince`, but I forgot to import it from foo.js. In fact, maybe I meant for `quince` to refer to something else, from a different module or even a global.)

And lastly in baz.js:

```js
import {quince} from "./foo";
import bar from "./bar";

export default function() {
  bar();
  console.log("baz", quince);
};
```

Rolling up baz.js:

```js
'use strict';

var quince = 42;

var foo = function() {
  console.log("foo");
};

var bar = function() {
  foo();
  console.log("bar", quince);
};

var baz = function() {
  bar();
  console.log("baz", quince);
};

module.exports = baz;
```

So, now bar.js’s `quince` is referring to the one in foo.js—but only because it was imported by baz.js. It seems like since bar.js refers to the symbol `quince`, the symbol imported from foo.js should be renamed to prevent masking (like `foo_quince` or whatever… though of course then you have to make sure `foo_quince` isn’t referred to anywhere in the original code, too).

**Comment by Rich-Harris** [comment]
Thanks - this is fixed in 0.7.5. Basically, names that are used in a module but not defined (or imported) are assumed to be globals, and those globals are [added to the list of names](https://github.com/rollup/rollup/blob/master/src/Bundle.js#L173-L178) that need to be deconflicted

**Comment by mbostock** [comment]
:+1: Thanks.
